vscode/src/circuit.ts

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { getCompilerWorker, log } from "qsharp-lang";
+import { Uri, window } from "vscode";
+import { basename, isQsharpDocument } from "./common";
+import { loadProject } from "./projectSystem";
+import type { IOperationInfo, VSDiagnostic } from "../../npm/lib/web/qsc_wasm";
+import { getTarget, getTargetFriendlyName } from "./config";
+import { sendMessageToPanel } from "./webviewPanel";
+import { getRandomGuid } from "./utils";
+import { EventType, UserFlowStatus, sendTelemetryEvent } from "./telemetry";
+
+const compilerRunTimeoutMs = 1000 * 60 * 5; // 5 minutes
+
+export async function showCircuitCommand(
```

Comment:
Looks good for now, but much of the logic here does seem to be boilerplate that's repeated for the Histogram and Estimates panel as well. We may want to refactor out the commonality at some point.

Reply:
I'd love to do that in a future PR. The error handling should probably be consolidated too - I print runtime errors into the webview now, but couldn't find a clean way to do it for the other panel (where would we even put that in the estimates?). Requires a bit of thought.